### ISSUE TITLE
Gradient calculation in run_inference and arcface_loss functions

### ISSUE THREAD

**Issue by alexhe101** [idonahum/photoVerse#18]
I have observed that the @torch.no_grad() decorator is used in the run_inference function, which means it will not compute gradients. What's more, I am also concerned about the arcface_loss function. This function takes in NumPy arrays as input and returns a loss value. My question is, since the input to the arcface_loss function is not a tensor that requires gradient calculation, will the resulting arcface_loss_val be able to backpropagate gradients during training?
Here's the code snippet for reference:

```
@torch.no_grad()
def run_inference(input_images, gen_images, face_analysis_func):
    ...

def arcface_loss(input_images, gen_images, face_analysis_func):
    arcface_scores = torch.tensor([cosine_similarity_between_images(gen_image, input_image, face_analysis_func) for gen_image, input_image in zip(gen_images, input_images)], dtype=torch.float32)
    loss = 1.0 - torch.abs(arcface_scores.mean())
    return loss

arcface_loss_val = arcface_loss(input_images, gen_images, face_analysis_func).to(device)
```
Will the gradients be calculated and backpropagated correctly in this setup?

**Comment by idonahum** [comment]
Hi, 
Thanks for input! 
This repo is still on setup, we find bugs all the time. I dont want to answer your question yet before we publish the final version, but you might be right about it. 

Thanks again.

**Comment by alexhe101** [comment]
> Hi, Thanks for input! This repo is still on setup, we find bugs all the time. I dont want to answer your question yet before we publish the final version, but you might be right about it.
> 
> Thanks again.

thank you for reply

**Comment by Haimzis** [comment]
Hi @alexhe101, Thank you for your input. 
We've fixed this.
